test/e2e/main_test.go

```diff
@@ -81,7 +89,22 @@ func testMain(m *testing.M) int {
 	}
 
 	rand.Seed(time.Now().UnixNano())
-	return m.Run()
+	if testOptions.enableCoverage {
+		if err := createDirForCoverage(testOptions.coverageDir); err != nil {
+			log.Fatalf("Error when creating coverage directory for coverageDir: %s, error: %v", testOptions.coverageDir, err)
+		}
+	}
+	ret := m.Run()
+	if testOptions.enableCoverage {
+		if err := testData.gracefulExitAntreaController(testOptions.coverageDir); err != nil {
+			log.Fatalf("Error when gracefully exit antrea controller: %v", err)
+		}
+		if err := testData.gracefulExitAntreaAgent(testOptions.coverageDir, "all"); err != nil {
+			log.Fatalf("Error when gracefully exit antrea agent: %v", err)
+		}
+
+	}
```

Comment:
nit: could we use a `defer` statement for this before calling `m.Run`, I think it would look nicer / more idiomatic

Reply:
good suggestion, defer statement is nicer.